backends/bmv2/backend.cpp

```diff
@@ -78,6 +79,86 @@ class ProcessControls : public BMV2::RemoveComplexExpressionsPolicy {
     }
 };
 
+/**
+This pass adds @name annotations to all fields of the user metadata
+structure so that they do not clash with fields of the headers
+structure.  This is necessary because both of them become global
+objects in the output json.
+*/
```

Comment:
Is this change specific to the one user metadata structure and the one headers structure in v1model.p4?

PSA adds half a dozen more top level parameters for top level controls that may be structs, all of which might have identical member names.  I guess this method can be extended for any number of such 'top level structs' ?

Reply:
We only care about structures which generate objects in the BMV2 toplevel namespace which can clash.